### ISSUE TITLE
'Cannot move' false positive when accessing the self pointer from struct functions

### ISSUE THREAD

**Issue by Finomnis** [intellij-rust/intellij-rust#4316]
## Environment

* **IntelliJ Rust plugin version:** 0.2.104.2132-192
* **Rust toolchain version:**  1.39.0-nightly (e44fdf979 2019-08-21) nightly-x86_64-pc-windows-msvc
* **IDE name and version:** IntellijIntelliJ IDEA 2019.2.1 (Community Edition)
* **Operating system:** Windows 10 Pro 1809

## Problem description
Accessing class member variables from within class methods causes the error `Cannot move`.

![intellij_rust_bug](https://user-images.githubusercontent.com/3129043/63804481-8b4cae80-c917-11e9-83a3-18e4577acc69.png)

## Steps to reproduce

Create a new Rust project and paste the following content into the `main.rs` file:
```
pub struct Test {
    x: usize,
}

impl Test {
    pub fn f(&self) {
        let _tmp = self.x;
    }
}

fn main() {
}
```

The code will compile fine, but the IDE will display `Cannot move` at the `self.x`.

## Potentially related bugs
https://github.com/intellij-rust/intellij-rust/issues/3062 - Seems to be the same problem but a year ago, was marked as closed back then
https://github.com/intellij-rust/intellij-rust/issues/4307 - Seems to be a similar bug in almost the same version

**Comment by Finomnis** [comment]
Disappeared after reboot ... causes unknown, not reproducible anymore. Happened directly after plugin update.

**Comment by Aloso** [comment]
@Finomnis I have this issue as well, please reopen it! Rebooting didn't fix it.

The error appears when dereferencing a `Copy` value:

```rust
struct Bar { x: i32 }

let bar = &Bar { x: 0 };
&bar.x;         // ok
bar.x;          // error

let x = &5;
*x;             // error
```

Plugin: 0.2.108.2148-192
IDE: IntelliJ Ultimate 2019.2.3
OS: Manjaro, Linux 5.3.6-1
Desktop: KDE Plasmas 5.16.5

**Comment by ortem** [comment]
@Aloso I've tried your example and it works fine for me. Do you have false-positive error annotations on exactly the same code?
<img width="366" alt="Screenshot 2019-10-30 at 17 21 11" src="https://user-images.githubusercontent.com/4854600/67866233-b0d57e80-fb39-11e9-901b-887ac954c968.png">

**Comment by dylemma** [comment]
I just started seeing this after upgrading to the `0.2.121.3059-201-stable` version of the plugin. Seems like a regression.
